### DISCUSSION TITLE
Creating and Deploying Custom Clients : npm run login

### DISCUSSION THREAD

**Discussion by shinhochoi** [mozilla/hubs#4653] [Hubs Cloud Questions] [answered]
Help me!

I launched my hubs cloud enterprise on the AWS EC2. 
Domain Name : khub1.com

I followed https://hubs.mozilla.com/docs/hubs-cloud-custom-clients.html. (it should be explained in more detail)
I did successfully forking and cloning, installing NodeJS on Windows 10, 
And executing  "npm ci"  is OK.

But "npm run login" ask  hostname.  ==> I entered my hubs cloud domain name 'khub1.com"
I am not sure my domain name "khub1.com" is correct answer for the hostname.

And I got the following error.
UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'os' of null ~~~

I do not know what the problem is and how to solve it.
I spent several days to solve this.

 Your reply and any guides would be helpful and I will appreciate it.
=============================
C:\shchoi\hubs>npm run login

> hubs@0.0.1 login C:\shchoi\hubs
> node -r @babel/register -r esm -r ./scripts/shim scripts/login.js

Logging into Hubs Cloud.

Host (eg hubs.mozilla.com): khub1.com
Phoenix Socket URL: wss://friendly-artificer.khub1.com
(node:8612) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'os' of null
    at os (C:\shchoi\hubs\src\utils/detect-safari.js:5:45)
    at new Store (C:\shchoi\hubs\src\storage/store.js:245:9)
    at C:\shchoi\hubs\scripts/login.js:38:17
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (C:\shchoi\hubs\scripts\login.js:19:103)
    at _next (C:\shchoi\hubs\scripts\login.js:21:194)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:8612) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:8612) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by pattersonbl2** [comment]
This would be a good question for our community discord. (https://discord.com/invite/dFJncWwHun)  I will try to run a custom client on my machine to see if I run into the same area.

**Reply by shinhochoi** [reply]
I am checking 'git' installation for the root cause. I am not familiar with git environment and am learning step by step of git. Thanks.

**Reply by shinhochoi** [reply]
I checked my 'git' installation on windows 10 and did successfully git commands such as git add/commit/push to update a sample text file to a forked hubs repository in the https://github.com/shinhochoi/hubs.
And I installed nodejs again and did npm ci.  I am stuck on the npm run login.  What to write for the host[eg hubs.mozilla.com):??????.   I don't know what I am doing wrong.

**Reply by nupur711** [reply]
what to write in host??

**Comment by brianpeiris** [comment, marked as answer]
This was fixed in the `hubs-cloud` branch with #4665. Please pull in the latest code from that branch and try again.

**Reply by shinhochoi** [reply]
Thanks brian, your quick reply helped me a lot.

I deleted my repository and forked again instead of git pull in for testing.  I did login successfully. I attached result here https://discord.com/channels/498741086295031808/890267483087446067/890673988408311829
